server/appServices.js

```diff
@@ -30,9 +46,16 @@ function updateAll(namespace, kubeclient) {
     .then(mobileclients => mobileclients.map(mobileclient => updateApp(namespace, mobileclient, kubeclient)))
     .then(promises => {
       updating = false;
+      connectionAttempts = 0;
       return Promise.all(promises);
     })
-    .catch(err => console.error(`Failed to update apps due to error`, err));
+    .catch(err => {
+      console.error(`Failed to update apps due to error`, err);
+
+      if (err.code === ECONNREFUSED) {
```

Comment:
If the error is not `ECONNREFUSED`, the stream is closed, but the app doesn't exits, since the `retryWatch` (containing the exit) is never called: in this case the MDC won't be restarted, but the `mobile-services.json` update will never work

Reply:
I see your point.

We know that `ERRCONNREFUSED` will close the streams (as tested). Do you suggest we do not add a condition here and call `retryWatch` whenever any error is caught here instead? 

My concern is that this might set up extra watches again when they are already open. One thing we could do is set a flag `isKubeApiDown=true` in `.x.on('close',` and instead of checking `ERRCONNREFUSED`, check if `isKubeApiDown` instead. What do you think?